### PR TITLE
Write tests

### PR DESCRIPTION
**Pull request by Thi-Ponce**
### Lesson objectives:
- Write unit tests with React Testing Library.
- Write unit tests with snapshots with Jest.

### CODE REVIEW

File: src/tests/calculator.test.jsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Calculator from '../pages/Calculator';
+
+it('renders correctly', () => {
+  const tree = renderer
+    .create(<Calculator />)
+    .toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (ahmetbozaci) [inline src/tests/calculator.test.jsx:10]: - [ ] Please fix the error.

image

File: src/tests/calculate.test.jsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import calculate from '../logic/calculate';
+
+it('when AC is selected, it should be null', () => {
+  const calculateObj = { total: 4, next: 2, operation: '+' };
+  const acBtn = 'AC';
+  const result = calculate(calculateObj, acBtn);
+  expect(result).toEqual({ total: null, next: null, operation: null });
+});
```
Comment (ahmetbozaci) [inline src/tests/calculate.test.jsx:8]: - [ ] I believe you can add a few more tests for the calculator file. Please add more tests.

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


 @Saadat123456 @Thi-Ponce  ✋🏼 
#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.



### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._
- [ ]  You should add a test to simulate user interaction. [Requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_tests.md#:~:text=Use%20React%20Testing%20Library%20to%20simulate%20user%20interaction.)

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost


___

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_


_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mentioning me as [ahmetbozaci](https://github.com/ahmetbozaci), I will receive a notification and will respond._
___
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Review by enaburekhan — APPROVED** [review]
Hi Saadat123456, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Code review by **Eric Enaburekhan**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**